### ISSUE TITLE
Microsoft formal objection to DOM 4.1 advancing to CR

### ISSUE THREAD

**Issue by michaelchampion** [w3c/dom#176]
Microsoft Formally Objects to the Web Platform WG Chair's resolution to the CfC on moving DOM 4.1 to Candidate Recommendation https://github.com/w3c/dom/issues/175#issuecomment-379754622 . We urge the Director not to approve a transition request, because taking DOM 4.1 to CR at this time is not justified in terms of the process and success criteria specified in the Web Platform WG charter.

First, we don’t believe the Chairs’ decision here is consistent with the charter’s Decision Policy https://www.w3.org/2017/08/webplatform-charter.html#decisions and the Process Document’s guidance https://www.w3.org/2018/Process-20180201/#Votes.    There has been virtually no attempt at “reaching consensus through technical discussion and compromise.”  Instead, the Chairs did a CfC (which got a bare majority of 14 thumbs up and 12 thumbs down), and declared consensus with no further discussion. It is true that the Process allows specs to advance without unanimous support, but that typically happens after extensive discussion and supermajority votes.  The Process also gives Chairs some leeway to make judgment calls, but we've never seen a spec advance after such a close vote with unanimous opposition by implementers.

Second, we don’t believe a CR transition is consistent with the Participation criteria in the charter https://www.w3.org/2017/08/webplatform-charter.html#participation. “To be successful, the Web Platform Working Group is expected to have 10 or more active participants for its duration, and to have the participation of industry leaders in fields relevant to the specifications it produces. If participants from fewer than three distinct browser-engine projects are participating in the group, its charter should be re-examined by the W3C.”  While these participation criteria are met for the WG as a whole, the DOM spec was not developed with participation from the “industry leaders” nor were three distinct DOM implementation projects engaged.   

Third, we don’t believe DOM 4.1 has been developed in a way consistent with the charter’s external liaisons section https://www.w3.org/2017/08/webplatform-charter.html#external-groups. There has not been any liaison with the WHATWG DOM editors. No explanation has been provided for the numerous substantive differences between the DOM Living Standard and the DOM 4.1 draft. There has been no reachout to minimize or mitigate the differences for web developers.  The W3C DOM editors may have valuable information about the real world interoperability of various features in the DOM Living Standard, but they have not been shared with the community.  Furthermore, the charter's rationale for re-working WHATWG specs doesn't apply here: “The specifications are developed with contributions from a broad range of stakeholders including implementors and authors, as well as specialists in accessibility, internationalisation, privacy, and security.”  From the evidence in the GitHub repository,  DOM 4.1 was developed by the editors, WG chairs , and Team  with little input from a broad range of stakeholders.  

To summarize: DOM 4.1 does not currently reflect the real consensus among implementers, developers, and various horizontal communities that successful W3C standards do.  On Microsoft’s behalf I urge the Director to return DOM 4.1 to the Working Group and broader community for real information sharing, consensus building, and difference minimization.  It should not advance on the Recommendation Track until that happens.

**Comment by othermaciej** [comment]
See also Apple's Formal Objection at https://github.com/w3c/dom/issues/175#issuecomment-380534425
